gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Jobs."""
+
+import six
+
+from gcloud.exceptions import NotFound
+from gcloud._helpers import _datetime_from_microseconds
+from gcloud.bigquery.table import SchemaField
+from gcloud.bigquery.table import _build_schema_resource
+from gcloud.bigquery.table import _parse_schema_resource
+
+
+class _LoadConfiguration(object):
+    """User-settable configuration options for load jobs."""
+    # None -> use server default.
+    _allow_jagged_rows = None
+    _allow_quoted_newlines = None
+    _create_disposition = None
+    _encoding = None
+    _field_delimiter = None
+    _ignore_unknown_values = None
+    _max_bad_records = None
+    _quote_character = None
+    _skip_leading_rows = None
+    _source_format = None
+    _write_disposition = None
+
+
+class LoadFromStorageJob(object):
+    """Asynchronous job for loading data into a BQ table from CloudStorage.
+
+    :type name: string
+    :param name: the name of the job
+
+    :type destination: :class:`gcloud.bigquery.table.Table`
+    :param destination: Table into which data is to be loaded.
+
+    :type source_uris: sequence of string
+    :param source_uris: URIs of data files to be loaded.
+
+    :type client: :class:`gcloud.bigquery.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the dataset (which requires a project).
+
+    :type schema: list of :class:`gcloud.bigquery.table.SchemaField`
+    :param schema: The job's schema
+    """
+    def __init__(self, name, destination, source_uris, client, schema=()):
+        self.name = name
+        self.destination = destination
+        self._client = client
+        self.source_uris = source_uris
+        self.schema = schema
+        self._properties = {}
+        self._configuration = _LoadConfiguration()
+
+    @property
+    def project(self):
+        """Project bound to the job.
+
+        :rtype: string
+        :returns: the project (derived from the client).
+        """
+        return self._client.project
+
+    @property
+    def path(self):
+        """URL path for the job's APIs.
+
+        :rtype: string
+        :returns: the path based on project and job name.
+        """
+        return '/projects/%s/jobs/%s' % (self.project, self.name)
```

Comment:
Given that it is only called when setting up an API call, you'll never be able to measure the difference:  KISS wins for me.